### PR TITLE
Uniform log messages

### PR DESCRIPTION
**Pull request by lexnv**
This PR normalizes the published log messages.

Errors are logged for any unrecoverable issues (ie, terminating a client/server
connection caused via error exit paths).

While at it, clean up some comments and documentation.

### CODE REVIEW

**Review by dvdplm — APPROVED** [review]

File: core/src/client/async_client/mod.rs [hunk 1]
```diff
@@ -551,11 +551,9 @@ async fn handle_backend_messages<S: TransportSenderT, R: TransportReceiverT>(
 			handle_recv_message(raw.as_ref(), manager, sender, max_notifs_per_subscription).await?;
 		}
 		Some(Err(e)) => {
-			tracing::error!("Error: {:?} terminating client", e);
 			return Err(Error::Transport(e.into()));
 		}
 		None => {
-			tracing::error!("[backend]: WebSocket receiver dropped; terminate client");
 			return Err(Error::Custom("WebSocket receiver dropped".into()));
```
Comment (niklasad1) [inline core/src/client/async_client/mod.rs:557]: can you fix this as well?

It is wrong as it has been moved to the `async client`

Comment (niklasad1) [inline core/src/client/async_client/mod.rs:557]:
```suggestion
			return Err(Error::Custom("TransportReceiver dropped".into()));
```

File: core/src/client/async_client/mod.rs [hunk 2]
```diff
@@ -692,14 +680,18 @@ async fn background_task<S, R>(
...
-						.await
-				{
-					tracing::warn!("{:?}", err);
-					let _ = front_error.send(err);
+				let frontend_value = if let Some(value) = frontend_value {
+					value
+				} else {
+					// User dropped the sender side of the channel.
+					// There is nothing to do just terminate.
+					tracing::debug!("[backend]: Frontend receiver dropped");
```
Comment (niklasad1) [inline core/src/client/async_client/mod.rs:688]:
```suggestion
					tracing::debug!("[backend]: client dropped");
```

**Review by niklasad1 — APPROVED** [review]

**Review by jsdw — APPROVED** [review]

File: core/src/client/async_client/mod.rs (hunk 1 above)
Comment (lexnv) [reply core/src/client/async_client/mod.rs]: This was solved by committing suggestions! Thanks!

**Pull request merged**